### DISCUSSION TITLE
Normalize skin weights possible?

### DISCUSSION THREAD

**Discussion by Olaf-Wolf3D** [donmccurdy/glTF-Transform#667] [Q&A] [answered]
Hey! Is there a way to normalize skin weights with gltf-transform? To get rid of  `ACCESSOR_WEIGHTS_NON_NORMALIZED` errors.
Limiting number of weights to top 4 most influential and then normalizing would also be great.
I didn't find a function for that in the docs.

**Comment by donmccurdy** [comment]
Hi @Olaf-Wolf3D! I did implement a function to normalize skin weights, though it is only used internally and not exported currently:

https://github.com/donmccurdy/glTF-Transform/blob/18c475741a35cc5c79af2617fcec4d2ac82bf1a8/packages/functions/src/quantize.ts#L434-L468

Sorting the weights (so the highest weights are all in WEIGHT_0) and then dropping WEIGHTS_1+ could be done very similarly. Feel free to open a 'feature request' issue or PR for this if you'd like these exported from the /functions package.

**Comment by donmccurdy** [comment, marked as answer]
Updating my previous answer for the latest release after #670 —

```javascript
const limit = 4;

for (const mesh of document.getRoot().listMeshes()) {
  for (const prim of mesh.listPrimitives()) {
    sortPrimitiveWeights(prim, limit);
  }
}
```
